core/src/main/java/org/lflang/generator/c/CGenerator.java

```diff
@@ -1947,6 +1947,15 @@ protected void setUpGeneralParameters() {
       // So that each separate compile knows about modal reactors, do this:
       CompileDefinitionsProperty.INSTANCE.update(targetConfig, Map.of("MODAL_REACTORS", "TRUE"));
     }
+    if (!targetConfig.get(
+        SingleThreadedProperty.INSTANCE)) { // FIXME: This logic is duplicated in CMake
+      pickScheduler();
+      // FIXME: this and pickScheduler should be combined.
```

Comment:
I'm a little worried about two FIXMEs just being copied instead of addressed properly.

Reply:
I had a look at the FIXMEs and removed them because the logic looks OK. I just made some simplifications.

Reply:
> I had a look at the FIXMEs and removed them because the logic looks OK. I just made some simplifications.
OK, sounds good!

I wasn't quite sure what duplication the FIXME was referring to...

Reply:
I checked it and I think we're good. It shouldn't have been left in there in the first place. It was probably my mistake.